packages/lesswrong/lib/collections/moderationTemplates/schema.ts

```diff
@@ -0,0 +1,51 @@
+import {schemaDefaultValue} from "../../collectionUtils";
+import SimpleSchema from "simpl-schema";
+
+const ALLOWABLE_COLLECTIONS: CollectionNameString[] = ['Messages', 'Comments'];
+
+type CollectionNameString = 'Messages' | 'Comments';
+
+const CollectionNameType = new SimpleSchema({
+  collectionName: {
+    type: String,
+    allowedValues: ALLOWABLE_COLLECTIONS,
+  }
+});
+
+const schema: SchemaType<DbModerationTemplate> = {
+  name: {
+    type: String,
+    canRead: ['guests'],
+    canCreate: ['members'],
+    canUpdate: ['members'],
+    order: 1,
+  },
+  collectionName: {
+    type: CollectionNameType.schema('collectionName'),
```

Comment:
So I discovered from an EA forum PR that this can just be `String`, and we can have a top-level `allowedValues` field, like this: `allowedValues: ALLOWABLE_COLLECTIONS`, which will both work correctly in terms of runtime validation and type code-gen.